tokio-postgres/src/lib.rs

```diff
@@ -246,8 +246,28 @@ pub enum SimpleQueryMessage {
     Row(SimpleQueryRow),
     /// A statement in the query has completed.
     ///
-    /// The number of rows modified or selected is returned.
-    CommandComplete(u64),
+    /// The command tag and number of rows modified or selected is returned.
+    CommandComplete(CommandTag),
+}
+
+/// Command Tag returned by the `SimpleQuery` stream.
+#[derive(Debug)]
+pub struct CommandTag(String);
+
+impl CommandTag {
+    pub(crate) fn new(tag: impl ToString) -> Self {
+        Self(tag.to_string())
+    }
+
+    /// The number of affected rows
+    pub fn rows(&self) -> u64 {
+        self.0.rsplit(' ').next().unwrap().parse().unwrap_or(0)
```

Comment:
Is there a reason that `CommandTag` was not defined as `CommandTag(String, Option<u64>)`?

The `unwrap()` in the above line will panic if there is no row count.

Reply:
> Is there a reason that CommandTag was not defined as CommandTag(String, Option<u64>)?

I think it's simpler to derive the rows count from the tag directly instead of calculating in advance.

It also reduces decisions I have to make.
For `SELECT 1`, if I store rows = Some(1), what do I put for the string? Do I keep `SELECT` or `SELECT 1`? 

The reason I'm not making the rows return an Option, is to preserve the existing convention in CommandComplete, that just keeps u64.
I don't know if I should change this behaviour to return an option.
It can be done of course.


> The unwrap() in the above line will panic if there is no row count.

Oops, I missed this. Let me see if I can fix that.

Reply:
@freshtonic , I removed the unwrap().
Do let me know if you think this should be abstracted differently.

Reply:
> I think it's simpler to derive the rows count from the tag directly instead of calculating in advance.

I honestly can't figure out what you mean here. There was previously a clear API (a `u64`) and now it's a string formatted in a particular way that contains a message and a fallible  row count. It's objectively not simple.

> It also reduces decisions I have to make.
> For SELECT 1, if I store rows = Some(1), what do I put for the string? Do I keep SELECT or SELECT 1?

More explanation required please @ragibkl . Before your change, the `CommandComplete(u64)` was being constructed without a tag , so presumably it has been modified by you to be constructed with a `String` instead now. 

Therefore I'm confused why you're confused about this:

> what do I put for the string? Do I keep SELECT or SELECT 1?

If the variant was defined as `CommandComplete(String, u64)` then you'd call `CommandComplete("SELECT", 1)`, right?